std/assembly/set.ts

```diff
@@ -108,6 +108,47 @@ export class Set<K> {
     }
   }
 
+  private getPtr(key: K): usize {
+    if (!this.has(key)) return EMPTY;
+    var oldPtr = changetype<usize>(this.entries) + HEADER_SIZE_AB;
+    var endPtr = oldPtr + <usize>this.entriesOffset * ENTRY_SIZE<K>();
+    while (oldPtr != endPtr) {
+      let entry = changetype<SetEntry<K>>(oldPtr);
+      if (!(entry.taggedNext & EMPTY) && entry.key == key) {
+        return oldPtr;
+      }
+      oldPtr += ENTRY_SIZE<K>();
+    }
+    return EMPTY;
+  }
+
+  forEach(callbackfn: (value1: K, value2: K, set: Set<K>) => void): void {
+    var visitedKeys = new Array<K>();
```

Comment:
Would be great if we could avoid instantiating another managed object here

Reply:
Ok, that's a good suggestion.